File: examples/companion-packages/swift-cocoapods-app/PawMatch/Charities.swift
// Curated static list of well-known animal-welfare charities. PawMatch never
// processes payments — every output includes a disclaimer reminding the user
// to verify ratings independently before giving.

import Foundation

struct Charity: Equatable {
    let slug: String
    let name: String
    let focus: String
    let description: String
    let url: String
    let ratingNote: String
}

enum Charities {
    static let disclaimer =
        "Curated example list — verify current ratings on Charity Navigator or GuideStar before giving."

    static let all: [Charity] = [
        Charity(slug: "best-friends", name: "Best Friends Animal Society",
                focus: "shelters",
                description: "No-kill movement; supports adoptions, shelters, and advocacy nationwide.",
                url: "https://bestfriends.org/donate",
                ratingNote: "Charity Navigator 4-star"),
        Charity(slug: "petsmart-charities", name: "PetSmart Charities",
                focus: "shelters",
                description: "Grants to local shelters; spay/neuter; adoption events.",
                url: "https://petsmartcharities.org/donate",
                ratingNote: "Charity Navigator 4-star (96% program ratio)"),
        Charity(slug: "brother-wolf", name: "Brother Wolf Animal Rescue",
                focus: "rescue",
                description: "Local rescue with national-impact outreach programs.",
                url: "https://bwar.org/donate",
                ratingNote: "Charity Navigator 4-star, GuideStar Platinum"),
        Charity(slug: "animal-welfare-institute", name: "Animal Welfare Institute",
                focus: "policy",
                description: "Policy and advocacy reducing cruelty inflicted on animals.",
                url: "https://awionline.org/donate",
                ratingNote: "Charity Navigator 4-star"),
        Charity(slug: "aspca", name: "ASPCA",
                focus: "shelters",
                description: "Adoption, anti-cruelty programs, and animal welfare advocacy.",
                url: "https://www.aspca.org/donate",
                ratingNote: "Charity Navigator 4-star"),
    ]

    static func findBySlug(_ slug: String) -> Charity? {
        all.first { $0.slug.caseInsensitiveCompare(slug) == .orderedSame }
    }

    static func filterByFocus(_ focus: String) -> [Charity] {
        if focus.caseInsensitiveCompare("all") == .orderedSame { return all }
        return all.filter { $0.focus.caseInsensitiveCompare(focus) == .orderedSame }
    }
}

File: examples/companion-packages/r-cran-app/R/charities.R
# pawmatch: animal welfare charity directory.

#' Charity directory.
#'
#' Curated, static list of well-known animal welfare organizations.
#' Example data — verify current ratings on Charity Navigator or
#' GuideStar before giving.
#' @export
pawmatch_charities <- function() {
  .charities_all
}

.make_charity <- function(slug, name, focus, description, url, rating_note) {
  list(
    slug = slug,
    name = name,
    focus = focus,
    description = description,
    url = url,
    rating_note = rating_note
  )
}

.charities_all <- list(
  .make_charity("best-friends", "Best Friends Animal Society", "shelters",
                "No-kill movement; supports adoptions, shelters, and advocacy nationwide.",
                "https://bestfriends.org/donate",
                "Charity Navigator 4-star"),
  .make_charity("petsmart-charities", "PetSmart Charities", "shelters",
                "Grants to local shelters; spay/neuter; adoption events.",
                "https://petsmartcharities.org/donate",
                "Charity Navigator 4-star (96% program ratio)"),
  .make_charity("brother-wolf", "Brother Wolf Animal Rescue", "rescue",
                "Local rescue with national-impact outreach programs.",
                "https://bwar.org/donate",
                "Charity Navigator 4-star, GuideStar Platinum"),
  .make_charity("animal-welfare-institute", "Animal Welfare Institute", "policy",
                "Policy and advocacy reducing cruelty inflicted on animals.",
                "https://awionline.org/donate",
                "Charity Navigator 4-star"),
  .make_charity("aspca", "ASPCA", "shelters",
                "Adoption, anti-cruelty programs, and animal welfare advocacy.",
                "https://www.aspca.org/donate",
                "Charity Navigator 4-star")
)

.charity_disclaimer <- paste0(
  "Curated example list — verify current ratings on Charity Navigator or ",
  "GuideStar before giving."
)

.charity_find <- function(slug) {
  if (is.null(slug) || !nzchar(slug)) {
    return(NULL)
  }
  target <- tolower(slug)
  for (c in .charities_all) {
    if (tolower(c$slug) == target) {
      return(c)
    }
  }
  NULL
}

.charity_filter_focus <- function(focus) {
  if (is.null(focus) || !nzchar(focus) || tolower(focus) == "all") {
    return(.charities_all)
  }
  target <- tolower(focus)
  Filter(function(c) tolower(c$focus) == target, .charities_all)
}
